src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.AddProperty.cs

```diff
@@ -135,6 +135,11 @@ internal static JsonPropertyInfo CreateProperty(Type declaredPropertyType, Type
                 args: null,
                 culture: null);
 
+            if (converter is JsonConverterFactory factory)
```

Comment:
Can this check be placed in here instead:
`internal JsonConverter DetermineConverterForProperty(Type parentClassType, Type runtimePropertyType, PropertyInfo propertyInfo)`

Reply:
Done